cpp/src/arrow/record_batch.h

```diff
@@ -80,6 +80,13 @@ class ARROW_EXPORT RecordBatch {
   /// in the resulting struct array.
   Result<std::shared_ptr<StructArray>> ToStructArray() const;
 
+  /// \brief Convert record batch with one data type to Tensor
+  ///
+  /// Create a Tensor object with shape (number of rows, number of columns) and
+  /// strides (type size in bytes, type size in bytes * number of rows).
+  /// Generated Tensor will have column-major layout.
+  Result<std::shared_ptr<Tensor>> ToTensor() const;
```

Comment:
```suggestion
  Result<std::shared_ptr<Tensor>> ToTensor(MemoryPool* pool = default_memory_pool()) const;
```

Should probably give this a memory pool argument (since it is allocating new memory), that can you pass through to AllocateBuffer

Reply:
Added it here: https://github.com/apache/arrow/pull/40064/commits/6a2c5bc850d171621e34bee09c5c53dd24d4df9b

Not sure how to test it though =)